### ISSUE TITLE
Elasticsearch bulk indexing module -  warning being thrown

### ISSUE THREAD

**Issue by girishgoudapatil** [ZJONSSON/node-etl#34]
There is a warning being thrown in elasticsearch bulk indexing module , may be a we don't need to return promise from execute function 

https://github.com/petkaantonov/bluebird/blob/master/docs/docs/warning-explanations.md#warning-a-promise-was-created-in-a-handler-but-was-not-returned-from-it

```
Warning: a promise was created in a handler but was not returned from it
    at execute (/node_modules/etl/lib/elasticsearch/bulk.js:77:20)
    at Bulk._fn (/node_modules/etl/lib/elasticsearch/bulk.js:119:10)
    at Bulk.Streamz._transform (/node_modules/etl/node_modules/streamz/streamz.js:97:16)
    at Bulk.Transform._read (_stream_transform.js:167:10)
    at Bulk.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:300:12)
    at clearBuffer (_stream_writable.js:407:7)
    at onwrite (_stream_writable.js:339:7)
    at WritableState.onwrite (_stream_writable.js:89:5)
    at afterTransform (_stream_transform.js:79:3)
    at TransformState.afterTransform (_stream_transform.js:54:12)
    at pop (/node_modules/etl/node_modules/streamz/streamz.js:79:29)
    at done (/node_modules/etl/node_modules/streamz/streamz.js:86:5)
```

**Comment by ZJONSSON** [comment]
Thanks for the report @girishgoudapatil .  I have not seen this error before and I would love to replicate.   I think we do need to return a Promise from the execute function as a way to control concurrent connections to ES, i.e. otherwise you might end up with far more connections to elasticsearch than intended.  (see more detail on concurrency and Promises within etl here: https://github.com/ZJONSSON/streamz)

As this error shows up within the `Promise.fromNode(..)` wrapper my initial hypothesis is that your elasticsearch client migh be creating  a `Promise` during the `bulk` statment that is not returned anywhere.   Can you elaborate on which Elasticsearch client (package and version) you are using?

**Comment by girishgoudapatil** [comment]
@ZJONSSON 

I am using elasticsearch@11.0.1 module , all the methods on elasticsearch client returns a promise , this might be causing the issue

**Comment by ZJONSSON** [comment]
Sorry for the late response - I think this has been fixed with https://github.com/ZJONSSON/node-etl/pull/41
published as etl@0.3.29.  @girishgoudapatil can you please reopen the issue if the problem persists?
